xlators/storage/posix/src/posix-common.c

```diff
@@ -622,6 +623,122 @@ posix_create_open_directory_based_fd(xlator_t *this, int pdirfd, char *dir_name)
             goto out;
         }
     }
+
+    if (sys_mkdirat(ret, "tt", 0700) < 0) {
+        if (errno != EEXIST) {
+            gf_msg(this->name, GF_LOG_ERROR, errno, P_MSG_HANDLE_CREATE,
+                   "Creating directory %s failed", "tt");
+            ret = -1;
+            goto out;
+        }
+    }
+
+out:
+    return ret;
+}
+
+int
+resolve_gfid_link_missed_during_rename(xlator_t *this, char *dirname)
+{
+    struct dirent scratch[2] = {
+        {
+            0,
+        },
+    };
+    struct posix_private *priv = this->private;
+    struct dirent *dentry = NULL;
+    DIR *drptr = NULL;
+    char tmp_gfid[64] = { 0, };
+    uuid_t exp_gfid = {
+        0,
+    };
+    uuid_t gfid_curr = {
+        0,
+    };
+    char dst_path[NAME_MAX] = {
+        0,
+    };
+    char gfid_link_path[NAME_MAX*2] = {
+        0,
+    };
+    int unlink_flag = 0;
+    int index = 0;
+    int ret = 0;
+    int dentry_len = 0;
+    int size = 0;
+
+    drptr = sys_opendir(dirname);
+    if (!drptr) {
+        gf_log(this->name, GF_LOG_ERROR,
+               "opendir is failed for special directory %s ", dirname);
+        ret = -1;
+        goto out;
+    }
+
+    while ((dentry = sys_readdir(drptr, scratch)) != NULL) {
+        dentry_len = strlen(dentry->d_name);
+        unlink_flag = 1;
+        if (dentry_len < UUID_CANONICAL_FORM_LEN)
+            continue;
+        /*if dentry_name has a rename suffix it means rename dir
+          fop was not succeeded completly so need to validate gfid and
+          heal if it was not linked properly
+        */
+        snprintf(tmp_gfid, (dentry_len - UUID_CANONICAL_FORM_LEN),
```

Comment:
this snprintf() and the next line (gf_uuid_parse() function) can be moved under the 'if' statement in line 703, no?

Reply:
done